batch/batch/worker/worker.py

```diff
@@ -2771,6 +2788,10 @@ async def run_job(self, job):
             raise
         except JVMCreationError:
             self.stop_event.set()
+        except IncompleteJVMCleanupError:
+            assert isinstance(job, JVMJob)
+            await self.recreate_jvm(job.jvm)
+            log.exception(f'while running {job}, incomplete cleanup from {job.jvm}')
```

Comment:
This results in two exceptional logs for a single exception. Let's either not log here or not log near the raise.